pkg/resources/apiserver/service.go

```diff
@@ -87,7 +87,10 @@ func ServiceReconciler(exposeStrategy kubermaticv1.ExposeStrategy, externalURL s
 			se.Spec.Ports[0].Port = 443
 			if exposeStrategy == kubermaticv1.ExposeStrategyTunneling {
 				se.Spec.Ports[0].TargetPort = intstr.FromInt(resources.APIServerSecurePort)
-				se.Spec.Ports[0].NodePort = 0 // allows switching from other expose strategies
+				// skip setting nodeport as zero if the service type is LoadBalancer
+				if se.Spec.Type != *apiServerServiceType {
+					se.Spec.Ports[0].NodePort = 0 // allows switching from other expose strategies
+				}
```

Comment:
`apiServerServiceType` can be nil. Also in line 67 we set the Service's, so this comparison can never be true.